packages/x-components/tests/unit/useDevice.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { mount } from 'cypress/vue2';
+import { createUseDevice } from '../../src/composables/create-use-device.composable';
+
+/**
+ * Mounts a test component displaying all the values of the composable created with
+ * `createUseDevice`.
+ *
+ * @param devices - The configuration of the devices breakpoints.
+ * @returns An API to test the component.
+ */
+function mountTestComponent<Device extends string>(
+  devices: Record<Device, number>
+): MountTestComponentAPI {
+  const useDevice = createUseDevice(devices);
+  mount({
+    setup() {
+      return {
+        devices: useDevice()
+      };
+    },
+    template: `
+        <div>
+          <div v-for="(value, device) in devices" :key="device">
+            {{ device }}: 
+            <span :data-test="device">{{ value.value }}</span>
+          </div>
+        </div>
+      `
+  });
+
+  return {
+    getDeviceValue(key: string) {
+      return cy.getByDataTest(key);
+    }
+  };
+}
+
+const defaultDevices = { mobile: 0, tablet: 744, desktop: 1200 };
+describe('testing createUseDevice composable factory', () => {
+  it('detects the right device', () => {
+    const { getDeviceValue } = mountTestComponent(defaultDevices);
```

Comment:
Should we test the orientation? I know that we are using the composable from vue use, but we should check our own values

Reply:
I would like, and I tried, but unfortunately, this feature only works on actual mobile/tablet/with-gyroscope devices. with cypress, you only can modify the size of the window but it always returns the "landscape" value, even in a "portrait" size.